ingestion/src/metadata/ingestion/source/dashboard/metabase.py

```diff
@@ -204,27 +200,43 @@ def get_lineage(self, dashboard_details: dict) -> AddLineageRequest:
                 resp_tables = self.req_get(f"/api/table/{chart_details['table_id']}")
                 if resp_tables.status_code == 200:
                     table = resp_tables.json()
-                    table_fqn = f"{self.source_config.dbServiceName}.{table['schema']}.{table['name']}"
-                    dashboard_fqn = f"{self.service.name}.{quote(dashboard_name)}"
-                    table_entity = metadata.get_by_name(entity=Table, fqn=table_fqn)
-                    chart_entity = metadata.get_by_name(
-                        entity=Model_Dashboard, fqn=dashboard_fqn
+                    from_fqn = fqn.build(
+                        self.metadata,
+                        entity_type=Table,
+                        service_name=self.source_config.dbServiceName,
+                        database_name=table.get("db").get("details").get("db"),
```

Comment:
chaining `get` might not always be the best approach. ` get` call is fine for a single element, so that we can control manually `None` values which would otherwise raise a `KeyError`.

Chaining 3 calls though gives no benefit, as it would just die for a `NoneType has no attribute get` or something similar. In these cases, having a KeyError (if we indeed want things to fail) would be an easier way to figure out what is going on

Reply:
Ok ok. So here I should directly access the dict elements like `table["db"]["details"]["db"]` right?

Reply:
yep, because otherwise we cannot be picking the right data. Thanks for the fix!